core/src/main/java/org/apache/accumulo/core/rpc/clients/TServerClient.java

```diff
@@ -54,6 +54,8 @@
 
 public interface TServerClient<C extends TServiceClient> {
 
+  static final String PREFERRED_HOST = "org.apache.accumulo.client.rpc.preferred.host";
+
   Pair<String,C> getThriftServerConnection(ClientContext context, boolean preferCachedConnections)
```

Comment:
Could add an IT that directly calls this method a few times with and without the preferred host prop set and see what it returns.